ThirdSemester/Lazy/Lazy/ILazy.cs

```diff
@@ -0,0 +1,17 @@
+﻿// <copyright file="ILazy.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
+
+/// <summary>
+/// Interface for lazy evaluation of a value.
+/// </summary>
+/// <typeparam name="T">The type of the computed value.</typeparam>
+public interface ILazy<T>
```

Comment:
```suggestion
public interface ILazy<out T>
```
Если можно указать вариантность, лучше указать, это упростит пользование контейнером.